src/main/java/com/bnkrll/controller/SessionController.java

```diff
@@ -0,0 +1,37 @@
+package com.bnkrll.controller;
+
+import com.bnkrll.model.Session;
+import com.bnkrll.service.SessionRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/session")
+@Slf4j
+public class SessionController {
+
+    private final SessionRepository sessionRepository;
+
+
+    public SessionController(@Qualifier("sessionRepositorySimulator")
+                                     SessionRepository sessionRepository) {
+        this.sessionRepository = sessionRepository;
+    }
+
+    @PostMapping
+    public void saveSession(@RequestBody Session session){
+        sessionRepository.save(session);
+    }
+
+
+    @GetMapping("/pastsessions")
```

Comment:
remove /pastsessions.
The resource is session. You are getting a list of sessions. GET /api/session?numOfSessions=30
I was suggesting something simple like last, but if you think numOfSessions is more readable then I can't disagree with that.

Reply:
Made the suggested changes

`    @GetMapping
    public List<Session> getPreviousSessions(
            @RequestParam(value = "last", defaultValue = "30")
                    int last){
        List<Session> previousSessions = sessionRepository.getLastSessions(last);
        return previousSessions;
    }`